classes/local/datasource/customfield.php

```diff
@@ -0,0 +1,123 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
```

Comment:
Let’s just call this ‘fields’ to be simpler eg fields.color fields.total fields.address reads naturally when using the expressions

Reply:
customfield now uses the shortname 'fields'.